src/components/OutputsTable/OutputCell/ErrorHandler.tsx

```diff
@@ -0,0 +1,94 @@
+import { type ModelOutput } from "@prisma/client";
+import { HStack, VStack, Text, Button, Icon } from "@chakra-ui/react";
+import { useEffect, useMemo, useState } from "react";
+import { BsArrowClockwise } from "react-icons/bs";
+import { rateLimitErrorMessage } from "~/sharedStrings";
+
+const MAX_AUTO_RETRIES = 3;
+
+export const ErrorHandler = ({
+  output,
+  refetchOutput,
+  numPreviousTries,
+}: {
+  output: ModelOutput;
+  refetchOutput: () => void;
+  numPreviousTries: number;
+}) => {
+  const [msToWait, setMsToWait] = useState(0);
+  const shouldAutoRetry =
+    output.errorMessage === rateLimitErrorMessage && numPreviousTries < MAX_AUTO_RETRIES;
+
+  const errorMessage = useMemo(() => breakLongWords(output.errorMessage), [output.errorMessage]);
+
+  useEffect(() => {
+    if (!shouldAutoRetry) return;
+    
+    let remainingTime = calculateDelay(numPreviousTries);
+    setMsToWait(remainingTime);
+    
+    const interval = setInterval(() => {
+      remainingTime -= 1000;
+      setMsToWait(remainingTime);
+      
+      if (remainingTime <= 0) {
+        refetchOutput();
+        clearInterval(interval);
+      }
+    }, 1000);
+  
+    return () => clearInterval(interval);
+  }, [shouldAutoRetry, setMsToWait, refetchOutput, numPreviousTries]);
+  
+
+  return (
+    <VStack w="full">
+      <HStack w="full" alignItems="flex-start" justifyContent="space-between">
+        <Text color="red.600" fontWeight="bold">
+          Error
+        </Text>
+        <Button
+          size="xs"
+          w={4}
+          h={4}
+          px={4}
+          py={4}
+          minW={0}
+          borderRadius={8}
+          variant="ghost"
+          cursor="pointer"
+          onClick={refetchOutput}
+          aria-label="refetch output"
+        >
+          <Icon as={BsArrowClockwise} boxSize={6} />
+        </Button>
+      </HStack>
+      <Text color="red.600">{errorMessage}</Text>
+      {msToWait > 0 && (
+        <Text color="red.600" fontSize="sm">
+          Retrying in {Math.ceil(msToWait / 1000)} seconds...
+        </Text>
+      )}
+    </VStack>
+  );
+};
+
+function breakLongWords(str: string | null): string {
```

Comment:
I assume this is so the text doesn't escape the container? using CSS for this like `word-break: break-all` handle that case more simply? https://css-tricks.com/almanac/properties/w/word-break/#aa-values

Reply:
break-all can break normal words as well, which we don't want